programs/storage_api/src/storage_instruction.rs

```diff
@@ -37,13 +37,49 @@ pub enum StorageInstruction {
     /// Expects 1 Account:
     ///    0 - Storage account with credits to redeem
     ///    1 - MiningPool account to redeem credits from
+    ///    2 - Replicator account to credit - this account *must* be the owner
     ClaimStorageReward,
     ProofValidation {
+        /// The segment during which this proof was generated
         segment: u64,
-        proofs: Vec<(Pubkey, Vec<CheckedProof>)>,
+        /// A Vec of proof masks per keyed replicator account loaded by the instruction
+        proofs: Vec<Vec<ProofStatus>>,
     },
 }
 
+fn get_ratios() -> (u64, u64) {
+    // max number bytes available for account metas and proofs
+    // The maximum transaction size is == `PACKET_DATA_SIZE` (1232 bytes)
+    // There are approx. 900 bytes left over after the storage instruction is wrapped into
+    // a signed transaction.
+    static MAX_BYTES: u64 = 900;
```

Comment:
Should we do something like `PACKET_DATA_SIZE - 300`

Reply:
Yeah maybe that is better. I'll fix it the next time I push something.